### ISSUE TITLE
MD023 (Headers must start at the beginning of the line) should be ignored within code blocks

### ISSUE THREAD

**Issue by astorije** [markdownlint/markdownlint#144]
Consider the following:

``````
- Something something:

  ```rb
  # A great comment
  foo = bar
  ```
``````

This will trigger the following:

```
[...]: MD023 Headers must start at the beginning of the line
```

In Ruby, Python, YAML, Git config, etc., comments start with a `#`, so [rule MD023](https://github.com/mivok/markdownlint/blob/master/docs/RULES.md#md023---headers-must-start-at-the-beginning-of-the-line) should be disabled in code blocks.

**Comment by mivok** [comment]
I'm not able to reproduce this in the current master of markdownlint. However, I did find a related issue that is fixed in a0cb406 (and should also fix the issue you're seeing if it turns out I was reproducing it wrong somehow). I have also added your issue as a test case so it should be caught in tests if it does show up again.

**Comment by christopheradams** [comment]
I can reproduce this exact issue on 0.4.0 and on master. This test suite passes with no failures, but if I run:

``` sh
bin/mdl test/rule_tests/headers_with_spaces_at_the_beginning.md
```

I get:

```
test/rule_tests/headers_with_spaces_at_the_beginning.md:3: MD023 Headers must start at the beginning of the line
test/rule_tests/headers_with_spaces_at_the_beginning.md:5: MD023 Headers must start at the beginning of the line
test/rule_tests/headers_with_spaces_at_the_beginning.md:8: MD023 Headers must start at the beginning of the line
```
